processing/src/main/java/org/apache/druid/query/GroupByMergedQueryRunner.java

```diff
@@ -114,10 +113,10 @@ public Void call()
                           {
                             try {
                               if (bySegment) {
-                                input.run(threadSafeQueryPlus, responseContext)
+                                input.run(queryPlus.withQueryMetricsCopied(), responseContext)
```

Comment:
Doesn't this allow silent metric skipping bugs? They could be recorded in the copied queryMetrics, but not finally reported in the "root" queryMetrics.

Reply:
Previously there were no QueryMetrics object in threadSafeQueryPlus, it will throw NPE if somebody tries to emit some metrics by mistake.

Reply:
OK, I'll revert changes relating to QueryPlus class. However, since QueryMetrics is now a thread-safe class, the old name withoutThreadUnsafeState() is out of date. I will use withoutQueryMetrics() instead.